src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +39,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(index -> index % 2 == 0 ? numbers.get(index) : numbers.get(index) - 1)
+                .filter(number -> number % 2 != 0)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException("Can't get average value from "
+                        + "list: < Here is our input " + numbers + " >"));
```

Comment:
```suggestion
                .getAsDouble();
```
As stated in common mistakes